cms/server/admin/templates/contest_tasks.html

```
@@ -63,43 +63,44 @@ <h1 class="title is-spaced">Tasks</h1>
 {% endif %}
 
-    <!-- Add Task Card -->
-    <div class="tp-filter-card">
-        <label class="tp-filter-label" for="task_id">Add Task:</label>
-        <form id="add-task-form" action="{{ url("contest", contest.id, "tasks", "add") }}" method="POST" style="display: flex; gap: 12px; flex: 1;">
+    <!-- Toolbar: Search + Add -->
+    {{ task_search_toolbar(contest.get_tasks()|length, admin.permission_all and unassigned_tasks|length > 0) }}
+
+    {% if admin.permission_all and unassigned_tasks|length > 0 %}
+    <div class="tp-filter-card mb-4" id="add-task-panel" {% if contest.get_tasks()|length > 0 %}style="display:none;"{% endif %}>
+        <form id="add-task-form" class="is-flex is-align-items-center is-flex-wrap-wrap" style="gap: 12px; flex: 1;" action="{{ url("contest", contest.id, "tasks", "add") }}" method="POST">
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ < ].

(spec-char-escape)